### ISSUE TITLE
Show why snake_case errors is emitted on single word variables (might be too small)

### ISSUE THREAD

**Issue by rmullin7286** [pylint-dev/pylint#2018]
### Steps to reproduce
    no = 'No'

### Current behavior
I'm getting an error saying that 'no' doesn't conform the camel_case style, even though it's only one word.

### Expected behavior
There should be no error.

### pylint --version output
pylint 1.8.4,
astroid 1.6.3
Python 3.6.3 (v3.6.3:2c5fed8, Oct  3 2017, 18:11:49) [MSC v.1900 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Hi, thanks for creating this issue! Can you also post your `pylintrc` file? You could also run pylint with `--include-naming-hint=y`, which will offer more indication on why it considers it a bad name. I think it might be because the name it's too short, by default we expect words of length 3+.

**Comment by eight04** [comment]
Is there a way to configure the minimum length of the variable name?

**Comment by PCManticore** [comment]
@eight04 Not quite, the only thing you could do is to change the `variable-rgx` to conform to the format and length you require. It might be a good idea to just separate the length of a variable/name in pylint from the format of a variable/name.

**Comment by gitaarik** [comment]
I tried setting [`variable-rgx`](http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=variable-rgx), but the `invalid-name` `snake_case` error doesn't go away. The `variable-rgx` setting does work for variables that really don't conform to the regex, but it doesn't take away the first warning, although the docs say that it should override the `variable-naming-style` setting. When I try to unset it with: `variable-naming-style=` then both errors go away. All by all, it doesn't make any difference for me.

**Comment by filips123** [comment]
Why is variable length important at all?
Variable can have good name even if it is short. Example for this can be `no` and some physical variables and constants.
What else can be done excluding changing `variable-rgx`? One name that conform this style is `var_no`, but is even worse then just `no`.

**Comment by pedro-w** [comment]
I agree with @filips123  - I've been using Python to do physics calculations and I use vars like `m`, `x` and `y` to match the equations. It looks strange and wordy to have to rename them `mass`, `x_coordinate` etc.
One option is to use the `good-names=` config item but it's a bit tedious to maintain in practice.

**Comment by PCManticore** [comment]
No, that's not possible.

**Comment by filips123** [comment]
Why? Can you add support for this? It would be useful if only one file contains that variables.

**Comment by PCManticore** [comment]
Because it requires changing how we do configuration, it's a high effort project that is not worth it for almost all the cases.

**Comment by matkoniecz** [comment]
To give more specific solution for people who will run into this bug: 

pylint t.py --variable-rgx=^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$ --argument-rgx=^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$

will stop complaining about perfectly good names. BTW, it would be nice to have solution similar to Rubocop config files - where it is easy to exclude complaints about names that make perfect sense (and some like `x` and `y` are predefined)

See https://github.com/matkoniecz/matkoniecz-ruby-style/commit/e2df9deba34054e86d40669416c716a3d67bfd51 (how people may configure allowed names) and https://github.com/rubocop-hq/rubocop/pull/6137 "Allow `db` to allowed names of `Naming/UncommunicativeMethodParamName` cop in default config"

**Comment by PCManticore** [comment]
@matkoniecz That's most likely an use case for [good-names](http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=good-names#basic-checker-options)

**Comment by blaiseli** [comment]
Reading the above discussion, I learned about the `good-names` option, and I'm grateful for it, but I'm not sure I understand whether something is already under way to change the way pylint reports about short variable names or not, and if yes, what will be the new behaviour.

I'm  using a freshly installed pylint (using pip under python3.7):
```
$ pylint --version
pylint 2.4.2
astroid 2.3.1
Python 3.7.2 (default, Mar  6 2019, 18:38:11) 
[GCC 4.9.0]
```

With this version, variable name `fh` gets caught as `Variable name "fh" doesn't conform to snake_case naming style`. `f_h` does not (which I indeed consider snake_case, so everything fine here), but `fhh` too does not trigger a message. This means that (fortunately) the use of underscore is not mandatory. So the problem with `fh` is not really that it is not snake_case, but that it is too short.

I may be remembering wrong, but it seems to me that some time ago, pylint would not report about `snake_case naming style` in the case of short variable names, but either more generally, about bad naming style, or more precisely, about the shortness of the name. I would argue the latter would be desirable as  new  behaviour. But maybe it is complicated to obtain if the names are checked against just one "one-size-fits-all" regexp.

**Comment by demongolem** [comment]
I just a clarification.  I thought it was standard name to use `f` for file objects and hence I am getting these errors.  Would there be a more acceptable name for a file object, or do I somehow have to deal with these myself?

**Comment by chrissshe** [comment]
@demongolem Reading above, I think you just need to `somehow have to deal with these myself`, which is really straightforward. 

Add `f` to `good-names`  See reference at http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=good-names#basic-checker-options

**Comment by chrissshe** [comment]
I got this warning even for stuff like `for i in range(100)` . Does pylint advise against using `i`?

**Comment by PCManticore** [comment]
@chrissshe Not by default, `i` is a permitted name among other, but you might have overwritten the default `good-names` list of permitted names if you are still getting this error for `i`.

**Comment by chrissshe** [comment]
@PCManticore That's probably the case. I edited `good-names` to allow `df (dataframe)`

**Comment by prjemian** [comment]
> Because it requires changing how we do configuration, it's a high effort project that is not worth it for almost all the cases.

I understand that it may be a high effort project, thus remain patient for it to stay on the long-term road map for implementation.  The remark about *not worth it for almost all the cases* is opinion and is at odds with a substantial community.  

In physics, common coding practice will often adopt names used in textbooks or published literature as being *best choice*.  These names are recognized internationally, even though they rely on domain-specific jargon, it is exactly the entirety of that community they reach.  

As we recommend value-added steps, such as linting, to coding practice for our non-programmer peers, we must advise them which of these findings to either heed or ignore.  For sure, none of these *pylint* novices will apply [local configurations](https://github.com/PyCQA/pylint/issues/2018#issuecomment-458069377) or [regexp rules](http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=good-names#basic-checker-options).  As it stands now, a pylint report is rife with such false positive findings.  We insist to our novices that these reports are only advisory.

A command-line option would be of great benefit to a significant community.

**Comment by PCManticore** [comment]
@prjemian Thanks for the comment, but I'm not sure why `--good-names` is not enough for you at this point? If `f`, `df` or whatever name you want to use still emits this error, simply passing `--good-names=f,df` to pylint should be enough to remove the error, without using a local configuration, nor regexp rule.

Also I'm not sure you're quoting me on the right context. The quoted answer was related to https://github.com/PyCQA/pylint/issues/2018#issuecomment-454777916, which suggests to add inline comments for `good-names` and similar options, which as far as I can tell, will not happen anytime soon, hence the too much effort comment of a change that is not worth doing.

I'm not sure what your comment is suggesting though. Is it the inline comment support? Is it something else? Right now this issue is about showing why a short variable might not match the naming style, namely because it is too short, not because it does not match the chosen naming style.

**Comment by prjemian** [comment]
> Because it requires changing how we do configuration, it's a high effort project that is not worth it for almost all the cases.

@PCManticore : This was your quote, right?

I missed that it referred to markup in the python file that identified potential good names.

----

Thanks to your reply, investigated further the `--good-names` option and agree that it will work.  Requires at least one pass to identify the flagged names, then exclude them, such as:

```
(py37) jemian@wow ~/.../projects/xpcs-nexus-8idi $ pylint load_xpcs_result.py
************* Module load_xpcs_result
load_xpcs_result.py:19:42: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:20:8: C0103: Variable name "Iq" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:23:8: C0103: Variable name "t0" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:25:8: C0103: Variable name "g2" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:27:8: C0103: Variable name "Int_2D" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:50:8: C0103: Variable name "ii" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:51:12: C0103: Variable name "jj" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:53:12: C0103: Variable name "ax" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:79:4: C0103: Variable name "QZ_colormap" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:82:9: C0103: Variable name "ax" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:83:4: C0103: Variable name "im" doesn't conform to snake_case naming style (invalid-name)
load_xpcs_result.py:99:10: C0103: Variable name "ax" doesn't conform to snake_case naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 7.89/10 (previous run: 10.00/10, -2.11)

(py37) jemian@wow ~/.../projects/xpcs-nexus-8idi $ pylint --good-names "Int_2D, f, Iq, t0, g2, ii, jj, ax, im, QZ_colormap" load_xpcs_result.py 

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 7.89/10, +2.11)

```

That's about as good as I can hope for.

**Comment by eraoul** [comment]
The definition of "snake_case" should have nothing to do with length. This bug should be fixed. It's hard for me to push for stricter usage of tools such as pylint when there are things like this causing problems. We tried to allow single-character variable names across the board, since we're using strict python 3 type hints. But even if I modify variable-rgx to allow short names, the snake_case error shows up anyway.

**Comment by alexchandel** [comment]
`--good-names` is insufficient because `pylint` is frequently invoked on large collections of files (including package files) by IDEs.  A setting to disable the minimum length of 3 is needed, and there's clearly enormous demand for this.

**Comment by dcsanx** [comment]
there are some names in the standard library that are camelCase eg like [maxDiff](https://docs.python.org/3.4/library/unittest.html#unittest.TestCase.maxDiff)
is the workaround to add these to 'good_names` ?

**Comment by graciejane** [comment]
> A setting to disable the minimum length of 3 is needed, and there's clearly enormous demand for this.

Completely agree with this.  A simple configuration option to set the minimum length will save a lot of annoying warnings and workarounds.  Or at least make a separate warning to say it's too short (so we can quickly ignore it) instead of not conforming to snake case which is misleading.  Also... do we really need a linter to tell us about the length of our variables?  It's not like it's something you can do unintentionally without noticing :)

**Comment by Toxe** [comment]
> Completely agree with this. A simple configuration option to set the minimum length will save a lot of annoying warnings and workarounds. Or at least make a separate warning to say it's too short (so we can quickly ignore it) instead of not conforming to snake case which is misleading.

Agreed. Just disable a "minimum length" option per default and let people choose what fits their needs best. No is forced to use it and this should make everyone happy.

**Comment by Havanero** [comment]
why does it say invalid-name: Class name "provider" doesn't conform to **PascalCase** naming style when is python?

**Comment by dreamalligator** [comment]
I did a quick check for what we have to update to make a PR for this. It looks like the tests in [`unittest_base.py`](https://github.com/PyCQA/pylint/blob/master/tests/checkers/unittest_base.py#L530-L546) are set up to entail more rewriting than you'd think than simply changing the [`SnakeCaseStyle`](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/base.py#L113-L124) regexes to allow for single characters. This is because the tests were written with the assumption that CamelCase and snake_case stylings will always be mutually exclusive.

To reproduce:

1. setup from https://github.com/PyCQA/pylint#testing

```bash
pip install tox pytest pytest-benchmark
```

2. Add a single character string to `SNAKE_CASE_NAMES`, such as

`unittest_base.py`:

```python
class TestNamePresets(unittest.TestCase):
    SNAKE_CASE_NAMES = {"tést_snake_case", "test_snake_case11", "test_https_200", "x"}
```

3. Run `pytest tests/checkers/unittest_base.py`.

And we start getting failures associated with CamelCase tests because the tests are assuming that CamelCase and snake_case styles cases are mutually exclusive. 

```bash
pytest tests/checkers/unittest_base.py
======================================================== test session starts ========================================================
platform linux -- Python 3.8.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/guardian452/projects/pylint, configfile: setup.cfg
plugins: benchmark-3.4.1
collected 25 items                                                                                                                  

tests/checkers/unittest_base.py .............s.....F.F..s                                                                     [100%]

============================================================= FAILURES ==============================================================
__________________________________________________ TestNamePresets.test_camel_case __________________________________________________

self = <checkers.unittest_base.TestNamePresets testMethod=test_camel_case>

    def test_camel_case(self):
        naming_style = base.CamelCaseStyle
    
        for name in self.CAMEL_CASE_NAMES:
            self._test_name_is_correct_for_all_name_types(naming_style, name)
        for name in self.ALL_NAMES - self.CAMEL_CASE_NAMES:
>           self._test_name_is_incorrect_for_all_name_types(naming_style, name)

tests/checkers/unittest_base.py:585: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/checkers/unittest_base.py:545: in _test_name_is_incorrect_for_all_name_types
    self._test_is_incorrect(naming_style, name, name_type)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

naming_style = <class 'pylint.checkers.base.CamelCaseStyle'>, name = 'x', name_type = 'class_const'

    @staticmethod
    def _test_is_incorrect(naming_style, name, name_type):
        rgx = naming_style.get_regex(name_type)
        fail = f"{name!r} not match pattern {rgx!r} (style: {naming_style}, type: {name_type})"
>       assert not rgx.match(name), fail
E       AssertionError: 'x' not match pattern re.compile('([^\\W\\dA-Z][^\\W_]*|__.*__)$') (style: <class 'pylint.checkers.base.CamelCaseStyle'>, type: class_const)
E       assert not <re.Match object; span=(0, 1), match='x'>
E        +  where <re.Match object; span=(0, 1), match='x'> = <built-in method match of re.Pattern object at 0x7fedc9f4c960>('x')
E        +    where <built-in method match of re.Pattern object at 0x7fedc9f4c960> = re.compile('([^\\W\\dA-Z][^\\W_]*|__.*__)$').match

tests/checkers/unittest_base.py:567: AssertionError
__________________________________________________ TestNamePresets.test_snake_case __________________________________________________

self = <checkers.unittest_base.TestNamePresets testMethod=test_snake_case>

    def test_snake_case(self):
        naming_style = base.SnakeCaseStyle
    
        for name in self.SNAKE_CASE_NAMES:
>           self._test_name_is_correct_for_all_name_types(naming_style, name)

tests/checkers/unittest_base.py:573: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/checkers/unittest_base.py:541: in _test_name_is_correct_for_all_name_types
    self._test_is_correct(naming_style, name, name_type)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

naming_style = <class 'pylint.checkers.base.SnakeCaseStyle'>, name = 'x', name_type = 'attr'

    @staticmethod
    def _test_is_correct(naming_style, name, name_type):
        rgx = naming_style.get_regex(name_type)
        fail = f"{name!r} does not match pattern {rgx!r} (style: {naming_style}, type: {name_type})"
>       assert rgx.match(name), fail
E       AssertionError: 'x' does not match pattern re.compile('([^\\W\\dA-Z][^\\WA-Z]{2,}|_[^\\WA-Z]*|__[^\\WA-Z\\d_][^\\WA-Z]+__)$') (style: <class 'pylint.checkers.base.SnakeCaseStyle'>, type: attr)
E       assert None
E        +  where None = <built-in method match of re.Pattern object at 0x7fedca331dc0>('x')
E        +    where <built-in method match of re.Pattern object at 0x7fedca331dc0> = re.compile('([^\\W\\dA-Z][^\\WA-Z]{2,}|_[^\\WA-Z]*|__[^\\WA-Z\\d_][^\\WA-Z]+__)$').match

tests/checkers/unittest_base.py:561: AssertionError
====================================================== short test summary info ======================================================
FAILED tests/checkers/unittest_base.py::TestNamePresets::test_camel_case - AssertionError: 'x' not match pattern re.compile('([^\\...
FAILED tests/checkers/unittest_base.py::TestNamePresets::test_snake_case - AssertionError: 'x' does not match pattern re.compile('...
============================================== 2 failed, 21 passed, 2 skipped in 0.23s ==============================================
```

**Comment by Nikolai-Hlubek** [comment]
> @pedro-w and @filips123 If `variable-rgx` or `good-names` is not good for you, you can also disable the check locally in places where you get this error. I don't think `x` `m` and similar variable names are readable to other people other than the one that wrote the code, so be mindful of this aspect as well. As it's mentioned in the documentation, `pylint` is not a one size fits it all, if something does not work for you, feel free to disable / adapt it to your needs.

I was currently evaluating if pylint is a nice addition to a GitLab CI workflow. 
But if I read such statements I'm having serious doubts about the direction of the project. A linter should come with sane defaults. Python has a big usage in data science and calculation algorithms that are encapsulated in functions ususally use just a letter or two. 
Does anybody really think the formulas given on this page have bad variable names?
https://en.wikipedia.org/wiki/Linear_interpolation

**Comment by Pierre-Sassoulas** [comment]
@Nikolai-Hlubek, first, @PCManticore is right that it's impossible to please everyone on naming, if we change this default, some web developer will get angry that we consider ``x0`` to be a proper name for a burger menu item, and will also have "serious doubt about the direction of the project" because as we all know python has "big usage in web development projects". For this reason it's not an easy fix where we just have to change the default : we need to create another checkers to separate the two things and not create a breaking changes on the existing configuration while doing it.

Secondly, this issue is labelled as high priority, but we're unpaid maintainers giving away their free time which explain that not everything is fixed right away. You're free to contribute if you want this to go faster.  Maybe you could create a sane default configuration for data science projects. Or add a bounty for this on IssueHunt.

**Comment by Nikolai-Hlubek** [comment]
Well you can and should read my comment as honest user feedback. It was not meant as insult. And please take my appology if you read it as such. 

Obviously I'm doing data analysis with python. 
I'm not big into linters, and was always happy with the one integrated into pycharm.
I tried to evaluate if pylint - which I used 15 years ago at university and had good memories about - would fit into a build pipeline. I got some strange linter errors as a result. One of which brought me here.
This error was so strikingly flawed combined with the snide remarks in this thread that I felt it worth to add a comment. 

Apparently pylint does not only try to enforce PEP8 but also a lot of other ideas. This doesn't work well with data analysis in a lot of cases (a topic for another day) and this is what I learned today. ;-) 
The idea of disabling a lot of options for the simple test I did is for me not practically feasible. Micromanaging the linter is too time consuming.

**Comment by Helltraitor** [comment]
Can I add one more case?

```Python
@dataclass
class Track(DownloadItem):
    """
    Track class represents a track and its album.

    Track class implements DownloadItem protocol, so it can be downloaded.
    """
    album: Album
    id_: Id
    info: dict[str, Any] = field(default_factory=dict)
    meta: dict[str, Any] = field(default_factory=dict)

    # v- This part is highlighted in PyCharm (I'm using standard external tool - Pylint (latest version, on Python 3.10.5))
    ###
    def id(self) -> int:
        """
        Retrieves id from Id object.

        Returns:
            int: The id value of track.
        """
        return self.id_.value
```

This is very strange to get such kind of warnings. If you check conform criterion by length that you should exclude all short words

**Comment by clavedeluna** [comment]
@Pierre-Sassoulas I'd be happy to help here since it's high priority, but given all the back and forth I can't figure out what the decision is here? What's the ask that would close this issue?

**Comment by Pierre-Sassoulas** [comment]
The short answer is that we need to create a new message ``name-too-short`` and refactor the way the current regex works. The latest discussion about specification is on this issue : https://github.com/PyCQA/pylint/issues/7305. I think the proposal in #7305 is pretty convincing, but would probably be easier if we wait for a major and do it as a breaking changes. With 3.0 approaching, the perfect time to push for it would be from now to the time when we release 3.0. Feel free to refurbish the discussion on the other issue. (We're not closing this one because it's one of the most up-voted issue in the repo and closing it would just hide that.)

**Comment by Avasam** [comment]
In the mean time, if you use Flake8, you can try out https://github.com/PyCQA/pep8-naming . Personally it fit my needs better than pylint for naming convention. Since I tend to follow PEP8 naming conventions anyway.

**Comment by dotkrnl** [comment]
Before this is resolved, for anyone who also wants to workaround this, you may use `good-names-rgxs = "^..?.?$"`

**Comment by ronaldroe** [comment]
There's a whole lot of discussion here _not_ addressing why a lower case, single word or letter variable would ever be flagged as _not_ snake case. The correct answer here is to update how you parse variable names and make it conform to what users expect and what they experience everywhere else. If this behavior were somehow preferable, it'd be a different story, but it's objectively _not_. It is unexpected, incorrect behavior and should be fixed, not worked around

**Comment by jacobtylerwalls** [comment]
I have a proposal I'd appreciate feedback on. I think we need a **fast fix** for this that does not require thinking through a new spec as discussed in #7305.

*Why?*
- Hiding a "short variable name" check inside the "name case" check is a major weakness for pylint, to the point that people are saying it's the reason they don't use the tool.
- Even if we put a band-aid on it by describing it correctly in the output message ("hey the name is too short!"), we'd be fixing the wrong problem, since we should be decoupling casing from length.
- This has been stuck for years.

In 3.0, if we just remove the length requirements from the default regexes, and explain in the release notes how to regain a check for length if wanted (via existing regexes), that would be _good enough_.

#2018 is a blocker for 3.0, I think. We can always do #7305 later. There is likely not time for significant new features in 3.0; we need to move to beta this summer.

*Is there an easy way for users to regain the 3+ character requirement if we change the default regexes?*
Yes!
- If users are okay with `disallowed-name` instead of `--invalid-name`, they can pass `bad-names-rgxs=^..?$`. That's short enough to suggest in the release note, and we can describe how it works, briefly. It would be more readable with an expression using `{1,2}`, but we'll need to fix #7229 first. (Good thing there's a PR: #7241).
- Or, if users really want `--invalid-name` emitted,  the users can copy the existing regexes from the old source into their configs. We can also show that in the release notes, or even spell out the old regexes.

***
Either way, for the user that wants pylint to continue to fail names like `x0`, it's not a half-a-day upgrade, and we can describe how to do it. It puts the burden of configuration in the smaller user group (people who want a min name length feature even though pylint nowhere advertises such a feature, and merely has one accidentally). And compared to some of the other "off by default" changes we make all the time, like moving `while-used` to an extension, it's moving in the right direction in terms of getting pylint less opinionated.

**Comment by DanielNoord** [comment]
> I have a proposal I'd appreciate feedback on. I think we need a **fast fix** for this that does not require thinking through a new spec as discussed in #7305.
> 
> _Why?_
> 
> * Hiding a "short variable name" check inside the "name case" check is a major weakness for pylint, to the point that people are saying it's the reason they don't use the tool.
> * Even if we put a band-aid on it by describing it correctly in the output message ("hey the name is too short!"), we'd be fixing the wrong problem, since we should be decoupling casing from length.
> * This has been stuck for years.
> 
> In 3.0, if we just remove the length requirements from the default regexes, and explain in the release notes how to regain a check for length if wanted (via existing regexes), that would be _good enough_.
> 
> #2018 is a blocker for 3.0, I think. We can always do #7305 later. There is likely not time for significant new features in 3.0; we need to move to beta this summer.
> 
> _Is there an easy way for users to regain the 3+ character requirement if we change the default regexes?_ Yes!
> 
> * If users are okay with `disallowed-name` instead of `invalid-name`, they can pass `bad-names-rgxs=^..?$`. That's short enough to suggest in the release note, and we can describe how it works, briefly. It would be more readable with an expression using `{1,2}`, but we'll need to fix [bad-names-rgxs mangles regular expressions with commas #7229](https://github.com/pylint-dev/pylint/issues/7229) first. (Good thing there's a PR: [Ignore quantifiers when splitting comma-separated regexes #7241](https://github.com/pylint-dev/pylint/pull/7241)).
> * Or, if users really want `--invalid-name` emitted,  the users can copy the existing regexes from the old source into their configs. We can also show that in the release notes, or even spell out the old regexes.
> 
> Either way, for the user that wants pylint to continue to fail names like `x0`, it's not a half-a-day upgrade, and we can describe how to do it. It puts the burden of configuration in the smaller user group (people who want a min name length feature even though pylint nowhere advertises such a feature, and merely has one accidentally). And compared to some of the other "off by default" changes we make all the time, like moving `while-used` to an extension, it's moving in the right direction in terms of getting pylint less opinionated.

Sounds good to me! Are you also working on a PR, I'd be happy to help with review. I think (like you said) this can be put clearly in the changelog, but we might also want to put this info a little bit more explicitly in the documentation. Can just be in the `details.rst` of the message as that is likely where people will end up looking for this.

**Comment by jacobtylerwalls** [comment]
I'll put up a PR very shortly

**Comment by dreamalligator** [comment]
amazing 👏
